src/pages/_app.tsx

```diff
@@ -10,11 +10,20 @@ import { FC } from 'react'
 import { ThemeProvider } from 'styled-components'
 import { QueryClientProvider } from '@tanstack/react-query'
 import { queryClient } from '@/lib/react-query'
+import { deleteCookie, getCookie } from 'cookies-next'
 
 const excludeRoutes = ['/login', '/cadastro']
 
 const App: FC<AppProps> = ({ Component, pageProps, router }) => {
   const shouldRenderLayout = !excludeRoutes.includes(router.pathname)
+
+  if (getCookie('dont_remember')) {
```

Comment:
O ideal aqui é esse `if` ser colocado dentro do useEffect para que ele não seja executado em todo re-render do componente. Lembrando também, que ao registrar um evento na inicialização, se faz necessário remover o mesmo no unmount do componente
Artigo interessante para leitura: [Memory Leak in React](https://medium.com/@shriharim006/memory-leaks-in-react-203de733f7d2)

```suggestion
  const removeCookie = () => {
      deleteCookie('@mentores-soujunior-v1:token')
      deleteCookie('dont_remember')
    }
    useEffect(() => {
      if (getCookie('dont_remember')) {
        window.addEventListener('beforeunload', removeCookie );
      }
      return () => {
        window.removeEventListener('beforeunload', removeCookie );
      }
    }, [])